examples/acoustic/Acoustic_codegen.py

```diff
@@ -55,33 +55,93 @@ def damp_boundary(damp, nbp):
+            # Create source symbol
+            p_src = Dimension('p_src', size=nsrc)
+            src = SourceLike(name="src", dimensions=[time, p_src], npoint=nsrc, nt=nt,
+                             dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                             coordinates=self.source.receiver_coords)
+            src.data[:] = self.source.traces[:]
+
+            # Create receiver symbol
+            p_rec = Dimension('p_rec', size=nrec)
+            rec = SourceLike(name="rec", dimensions=[time, p_rec], npoint=nrec, nt=nt,
+                             dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                             coordinates=self.data.receiver_coords)
+
+            # Create the forward wavefield
+            u = TimeData(name="u", shape=self.model.get_shape_comp(), time_dim=nt,
+                         time_order=2, space_order=self.s_order, save=False,
+                         dtype=self.damp.dtype)
+            fw = ForwardOperator(self.model, u, src, rec, self.damp, self.data,
                                  time_order=self.t_order, spc_order=self.s_order,
                                  profile=True, save=False, dse=dse, compiler=compiler)
             self.at = AutoTuner(fw)
             self.at.auto_tune_blocks(self.s_order + 1, self.s_order * 4 + 2)
 
-    def Forward(self, save=False, cache_blocking=None,
-                auto_tuning=False, dse='advanced', compiler=None, u_ini=None):
+    def Forward(self, save=False, cache_blocking=None, auto_tuning=False,
+                dse='advanced', compiler=None, u_ini=None, legacy=True):
+        nt, nrec = self.data.shape
+        nsrc = self.source.shape[1]
+        ndim = len(self.damp.shape)
+        h = self.model.get_spacing()
+        dtype = self.damp.dtype
+        nbpml = self.model.nbpml
+        dt = self.model.get_critical_dt()
+        if self.t_order == 4:
+            dt *= 1.73
+
+        # Create source symbol
+        p_src = Dimension('p_src', size=nsrc)
+        src = SourceLike(name="src", dimensions=[time, p_src], npoint=nsrc, nt=nt,
+                         dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                         coordinates=self.source.receiver_coords)
+        src.data[:] = self.source.traces[:]
+
+        # Create receiver symbol
+        p_rec = Dimension('p_rec', size=nrec)
+        rec = SourceLike(name="rec", dimensions=[time, p_rec], npoint=nrec, nt=nt,
+                         dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                         coordinates=self.data.receiver_coords)
 
         if auto_tuning:
             cache_blocking = self.at.block_size
-        fw = ForwardOperator(self.model, self.src, self.damp, self.data,
+
+        # Create the forward wavefield
+        u = TimeData(name="u", shape=self.model.get_shape_comp(), time_dim=nt,
+                     time_order=2, space_order=self.s_order, save=save,
+                     dtype=self.damp.dtype)
+        u.pad_time = save
```

Comment:
I don't think there is a test that explicitly verifies this. The legacy code will put the result of the first time step in `t=0`. Your new code, I believe, will put it in `t=2` for `time_order=2`. 
This will need to be manually verified which parts of the code expect the first timestep to be in `t=0` and which in `t=2`. 
Since the tests run for a small number of time steps, the difference this produces will not show up.